### DISCUSSION TITLE
Zoom panels keyboard shortcuts

### DISCUSSION THREAD

**Discussion by factorialmap** [posit-dev/positron#3746] [Q&A] [answered]
Hi everyone. Is there in Positron an option that allows you to expand the panels? 

Example: 
```
Keyboard shortcuts in RStudio. Tools>Modify keyboard shortcuts
1. Ctrl+Shift+2 = zoom console  | Ctrl+Shift+2 again =  back to normal screen
2. Ctrl+Shift+6 = zoom Plots      | Ctrl+Shift+6 again = back to normal screen
3. Ctrl+Shift+9 = zoom viewer    | Ctrl+Shift+9 again = back to normal screen
```
These shortcuts are helpful for quickly showing results 

![image](https://github.com/posit-dev/positron/assets/46722179/1f51afa7-1eb0-4146-8afe-d621c27325ec)

**Comment by juliasilge** [comment]
There are not yet commands to zoom to a view (i.e. have it take up the whole window), but there *are* commands available to focus on Plots, Variables, the Viewer, the Console, etc:

![Screenshot 2024-07-01 at 5 02 22 PM](https://github.com/posit-dev/positron/assets/12505835/acb24732-a7f1-4176-a927-fa5bb405604b)

These mostly don't currently have keyboard shortcuts but you can [set up your own customized ones](https://code.visualstudio.com/docs/getstarted/keybindings) according to your preferences.

We can use this discussion to gather feedback/interest in an RStudio-like zoom feature. Thanks!

**Comment by srhrnkn** [comment]
I’m definitely interested in zoom - was trying to figure out if there’s a good way to enlarge plots temporarily and couldn’t come up with anything.

**Reply by juliasilge** [reply, marked as answer]
For plots, we have #2270 where we are covering having a zoom or pop-out feature. We know how important this is and it will happen much sooner than a possible general RStudio-like zoom feature for all panes.

**Reply by juliasilge** [reply]
I started https://github.com/posit-dev/positron/issues/4179 to collect ideas on an RStudio-like zoom feature, if you would like to share your perspective.

**Comment by Kikkervelf** [comment]
I also love the shortcuts to zoom in on Help and Plots in RStudio. This is what I did in Positron to mimick it: press Ctrl+K Ctrl+S to go to the Keyboard Shortcuts in Positron. Then set the following three shortcuts:

**View: Toggle Secondary Side Bar Visibility** - Ctrl+Shift+1
**Help: Focus on Help View** - Ctrl+Shift+3
**Session: Focus on Plots View** - Ctrl+Shift+6

If there are any overlapping shortcuts, simply press the right mouse button and click "Remove Keybinding".

Now you have to maximize the plots pane with your cursor. Make it as large as you can.

Voila, now you can toggle freely between the help pane (Ctrl+Shift+3), plots pane (Ctrl+Shift+6) and the editor/console (Ctrl+Shift+1).

![afbeelding](https://github.com/user-attachments/assets/0bc00608-5378-402e-9e5b-1cad11da894a)

**Reply by factorialmap** [reply]
It works really good. Thanks for sharing this.
